src/main/java/org/apache/flink/scheduler/benchmark/topology/BuildExecutionGraphBenchmark.java

```diff
@@ -0,0 +1,90 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.scheduler.benchmark.topology;
+
+import org.apache.flink.runtime.executiongraph.ExecutionGraph;
+import org.apache.flink.runtime.jobgraph.JobGraph;
+import org.apache.flink.runtime.jobgraph.JobVertex;
+import org.apache.flink.scheduler.benchmark.ColdStartSchedulerBenchmarkBase;
+import org.apache.flink.scheduler.benchmark.JobConfiguration;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import java.util.List;
+
+import static org.apache.flink.scheduler.benchmark.SchedulerBenchmarkUtils.createDefaultJobVertices;
+import static org.apache.flink.scheduler.benchmark.SchedulerBenchmarkUtils.createExecutionGraph;
+import static org.apache.flink.scheduler.benchmark.SchedulerBenchmarkUtils.createJobGraph;
+
+
+public class BuildExecutionGraphBenchmark extends ColdStartSchedulerBenchmarkBase {
+
+	private JobGraph jobGraph;
+	private ExecutionGraph executionGraph;
+
+	@Param({"BATCH", "STREAMING"})
+	private JobConfiguration jobConfiguration;
+
+	public static void main(String[] args) throws RunnerException {
+		Options options = new OptionsBuilder()
+				.verbosity(VerboseMode.NORMAL)
+				.include(".*" + BuildExecutionGraphBenchmark.class.getCanonicalName() + ".*")
+				.build();
+
+		new Runner(options).run();
+	}
+
+	@Setup(Level.Iteration)
+	public void setupIteration() throws Exception {
+		final List<JobVertex> jobVertices = createDefaultJobVertices(
+				PARALLELISM,
+				jobConfiguration.distributionPattern,
+				jobConfiguration.resultPartitionType);
+		jobGraph = createJobGraph(
+				jobVertices,
+				jobConfiguration.scheduleMode,
+				jobConfiguration.executionMode);
+
+		executionGraph = createExecutionGraph(jobGraph);
+	}
+
+	@TearDown(Level.Iteration)
+	public void teardownIteration() {
+		jobGraph = null;
+		executionGraph = null;
+		System.gc();
+	}
+
+	@Benchmark
+	@BenchmarkMode(Mode.SingleShotTime)
+	public void buildTopology() throws Exception {
+		executionGraph.attachJobGraph(jobGraph.getVerticesSortedTopologicallyFromSources());
```

Comment:
can be `executionGraph.attachJobGraph((jobVertices)` and then no need to store the jobGraph